### DISCUSSION TITLE
Please let WebView support CSS media query ‘prefers-color-scheme'

### DISCUSSION THREAD

**Discussion by Amir000001** [dotnet/maui#22256] [Ideas]
### Description

For better user experience with dark mode on web sites, Please let WebView support CSS media query prefers-color-scheme.

### Public API Changes

Android & iOS both platforms have problems but different.
Android's WebView can not detect media query prefers-color-scheme at all.
iOS's WebView is correctly following OS's appearance, but there is white background flashed before page loaded under dark appearance.

Let WebView with CSS media query prefers-color-scheme support, so the pages implemented would switch theme by OS automatically.

### Intended Use-Case

MS offical doc site has implemented prefers-color-scheme
So everyone can test on https://docs.microsoft.com/en-us/xamarin/

**Comment by Eilon** [comment]
According to this post: https://stackoverflow.com/questions/57449900/letting-webview-on-android-work-with-prefers-color-scheme-dark

It sounds like on Android as long as you target the right API version it should all just work. Can you confirm whether you've tried that? Or what did you try and what doesn't work?

**Comment by ghost** [comment]
Hi @Amir000001. We have added the "s/needs-info" label to this issue, which indicates that we have an open question for you before we can take further action. This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

**Comment by Amir000001** [comment]
> Согласно этому сообщению: https://stackoverflow.com/questions/57449900/letting-webview-on-android-work-with-prefers-color-scheme-dark
> 
> Похоже, что на Android, если вы нацеливаете правильную версию API, все должно работать. Можете ли вы подтвердить, пробовали ли вы это? Или что вы пробовали и что не работает?

I tried (`@media (prefers-color-scheme: dark)) and (window.matchMedia("(prefers-color-scheme: dark)")`) and they both don't work, I wish it worked out of the box.
I found a workaround for myself:

```
        string webviewTheme = (Application.Current.RequestedTheme == AppTheme.Dark) ? "dark-theme" : "light-theme";
        WebPage.Html = $@"
<html>
<head>
<style>
body.light-theme{{background-color: #fcfffc; color: #343434;}}
body.dark-theme {{background-color: #333333; color: #fcfffc;}}
</style>
</head>
<body>
<h1> Xamarin.Forms </h1>
<p> Welcome to {value}.</p>
<script type=""text/javascript"">
document.body.classList.add(""{webviewTheme}"");
</script>
</body></html>";
```

But this does not work when the OS theme changes and the webview is open, you need to refresh the page
